File: Lib/Telerik/Examples/QuickStart/GridView/Rows/RowDetails/RowDetailsRowElement.cs
using System;
using Telerik.WinControls.UI;
using System.Drawing;
using Telerik.WinControls.Data;
using Telerik.WinControls;

namespace Telerik.Examples.WinControls.GridView.Rows.RowDetails
{
    public class RowDetailsRowElement : GridDataRowElement
    {
        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(GridDataRowElement);
            }
        }

        public override void UpdateInfo()
        {
            if (!this.RowInfo.IsCurrent)
            {
                this.RowInfo.Height = TableElement.RowHeight;
            }
            else
            {
                this.RowInfo.Height = ((RowDetailsGrid)this.GridControl).DetailsRowHeight;
            }
            base.UpdateInfo();
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/GridView/Rows/RowDetails/RowDetailsRowElement.vb
﻿Imports Telerik.WinControls.UI
Imports Telerik.WinControls.Data
Imports Telerik.WinControls

Namespace Telerik.Examples.WinControls.GridView.Rows.RowDetails
	Public Class RowDetailsRowElement
		Inherits GridDataRowElement
		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(GridDataRowElement)
			End Get
		End Property

		Public Overrides Sub UpdateInfo()
			If Not Me.RowInfo.IsCurrent Then
				Me.RowInfo.Height = TableElement.RowHeight
			Else
				Me.RowInfo.Height = (CType(Me.GridControl, RowDetailsGrid)).DetailsRowHeight
			End If
			MyBase.UpdateInfo()
		End Sub
	End Class
End Namespace